control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -154,6 +154,15 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	//get all custom httprouteFilters these routes reference
+	for _, h := range httpRoutes {
+		_, err := r.getExternalFiltersForHTTPRoute(ctx, h, resources)
```

Comment:
Does it make sense to drop this into the `getRelatedHTTPRoutes` function above? Don't really care either way, but just wondering if it makes more sense for organizational grouping sake.